botocore/docs/method.py

```diff
@@ -235,3 +238,8 @@ def document_model_driven_method(section, method_name, operation_model,
                 include=include_output, exclude=exclude_output)
     else:
         return_section.write(':returns: None')
+
+    # Add the shared examples
+    if shared_examples:
```

Comment:
I wonder if we should add this here because this could affect other (non-client) methods as this gets used for waiters, paginators, and resources if shared_examples gets set. I think it would be better if we just call this ``document_shared_examples`` method in the client as the examples do not pertain to the general case, just the client. I think we can bump it down to this method though once we get examples for waiters, paginators etc.